components/server/src/api/server.ts

```diff
@@ -211,16 +233,38 @@ export class API {
         };
     }
 
-    private async verify(context: HandlerContext) {
-        const user = await this.sessionHandler.verify(context.requestHeader.get("cookie") || "");
-        if (!user) {
+    private async verify(context: HandlerContext): Promise<string> {
+        const claims = await this.sessionHandler.verifyJWTCookie(context.requestHeader.get("cookie") || "");
+        const userId = claims?.sub;
+        if (!userId) {
             throw new ConnectError("unauthenticated", Code.Unauthenticated);
         }
-        const fgaChecksEnabled = await isFgaChecksEnabled(user.id);
+        const fgaChecksEnabled = await isFgaChecksEnabled(userId);
         if (!fgaChecksEnabled) {
             throw new ConnectError("unauthorized", Code.PermissionDenied);
         }
-        return user;
+        return userId;
+    }
+
+    private readonly rateLimiters = new Map<string, RateLimiterRedis>();
+    private getRateLimitter(options: IRateLimiterOptions): RateLimiterRedis {
+        const sortedKeys = Object.keys(options).sort();
+        const sortedObject: { [key: string]: any } = {};
+        for (const key of sortedKeys) {
+            sortedObject[key] = options[key as keyof IRateLimiterOptions];
+        }
+        const key = JSON.stringify(sortedObject);
+
+        let rateLimiter = this.rateLimiters.get(key);
```

Comment:
storing and fetching rate lmiters by options seems wrong. Should that not go by user and method?

Reply:
What is wrong? How user is relevant? If we have by each method we will create very many rate limiter instances unnecessary.

Reply:
I guess what we really want to cache here is the redis connection, but we can't do that so we cache by options?

Reply:
I think there is only on redis connection and each instance almost the same just apply different options. We can have a myriad of instances with the same options and same connections or we can have only one. I think it is also a reason why before we wanted to have groups to give identity to rate limiters before.

Reply:
ok, after checking the library it seems ok and common 🤷